### PR TITLE
DOC: Correct groupby().mean() usage in table layout getting started article

### LINKED ISSUE #55599: DOC: Getting started example on using groupby().mean() alongside pivot throws error
### Pandas version checks

- [X] I have checked that the issue still exists on the latest versions of the docs on `main` [here](https://pandas.pydata.org/docs/dev/)

### Location of the documentation

https://pandas.pydata.org/docs/dev/getting_started/intro_tutorials/07_reshape_table_layout.html

### Documentation problem

If you run this line discussing the connection between pivot tables and `groupby()`,
```python
air_quality.groupby(["parameter", "location"]).mean()
```
you actually get TypeErrors related to string conversions and the aggregation function failing.

### Suggested fix for documentation

Restricting to numeric values only shows the intended result that matches the pivot table. I.e.,
```python
print(air_quality.pivot_table(
    values="value",
    index="location",
    columns="parameter",
    aggfunc="mean",
    margins=True
))

print(air_quality.groupby(["parameter", "location"]).mean(numeric_only=True))
```
shows
```
parameter                 no2       pm25        All
location                                           
BETR801             26.950920  23.169492  24.982353
FR04014             29.374284        NaN  29.374284
London Westminster  29.740050  13.443568  21.491708
All                 29.430316  14.386849  24.222743
                                  value
parameter location                     
no2       BETR801             26.950920
          FR04014             29.374284
          London Westminster  29.740050
pm25      BETR801             23.169492
          London Westminster  13.443568
```

### PR DESCRIPTION
**Pull request by ragibson**
This would close #55599 in line with the suggestion in the issue.

### CODE REVIEW

**Review by rhshadrach — COMMENTED** [review]
Thanks for the PR!

File: doc/source/getting_started/intro_tutorials/07_reshape_table_layout.rst [hunk 1]
```diff
@@ -266,7 +266,7 @@ For more information about :meth:`~DataFrame.pivot_table`, see the user guide se
 
     ::
 
-        air_quality.groupby(["parameter", "location"]).mean()
+        air_quality.groupby(["parameter", "location"]).mean(numeric_only=True)
```
[the remark this answers is not in the document]
Comment (ragibson) [reply doc/source/getting_started/intro_tutorials/07_reshape_table_layout.rst:269]: That's fair -- I guess it is better to not depend on parameter and location being non-numeric.

**Review by rhshadrach — APPROVED** [review]
lgtm - ping on green.

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @ragibson

**Pull request merged**